iocore/net/SSLNetVConnection.cc

```diff
@@ -1325,6 +1325,8 @@ SSLNetVConnection::sslServerHandShakeEvent(int &err)
         if (!this->setSelectedProtocol(proto, len)) {
           return EVENT_ERROR;
         }
+        this->set_negotiated_protocol_id({reinterpret_cast<const char *>(proto), static_cast<size_t>(len)});
```

Comment:
`setSelectedProtocol` and `setSelectedProtocol` are provided by the same class and these both receive exactly the same parameters. Why do we have to call two functions? Is there any case we want to call only one of them?

Reply:
ALPN with Partial Blind Tunnel has a situation of that negotiated application protocol is different from the endpoint which is actually used.

Reply:
Hmm, registering a different endpoint for tunneling vc maybe more generic.